apollo-federation/src/query_graph/path_tree.rs

```diff
@@ -415,6 +459,14 @@ where
                     trigger: child.trigger.clone(),
                     conditions: merge_conditions(&child.conditions, &other_child.conditions),
                     tree: child.tree.merge(&other_child.tree),
+                    context_to_selection: merge_context_to_selection(
+                        &child.context_to_selection,
+                        &other_child.context_to_selection,
+                    ),
```

Comment:
This is somewhat inline with Renee's comment. If `context_to_selection` were not an `Option`, this could be replaced with the following. If `None` does have meaning here or is meant to communicate that these might not be used, ignore this.
```
                    context_to_selection: child
                      .context_to_selection
                      .iter()
                      .chain(&other_child.context_to_selection)
                      .collect()
```

Reply:
@TylerBloom It's a little confusing to have separate vectors for `edge_triggers`, `edge_conditions`, `context_to_selection`, and `parameter_to_context`, but the idea is that they are all of the same length and each index will translate to a single `GraphPathItem`. However a single `GraphPathItem` might have some of these values be None, hence the Option inside of the vector.